File: android/src/main/java/com/superwallreactnative/models/PaywallCloseReason.kt
package com.superwallreactnative.models

import com.superwall.sdk.paywall.presentation.PaywallCloseReason

fun PaywallCloseReason.rawValue(): String = when (this) {
  is PaywallCloseReason.SystemLogic -> "systemLogic"
  is PaywallCloseReason.ForNextPaywall -> "forNextPaywall"
  is PaywallCloseReason.WebViewFailedToLoad -> "webViewFailedToLoad"
  is PaywallCloseReason.ManualClose -> "manualClose"
  is PaywallCloseReason.None -> "none"
}

File: src/public/PaywallCloseReason.ts
export enum PaywallCloseReason {
  SystemLogic = "systemLogic",
  ForNextPaywall = "forNextPaywall",
  WebViewFailedToLoad = "webViewFailedToLoad",
  ManualClose = "manualClose",
  None = "none",
}

export namespace PaywallCloseReason {
  export function toJson(reason: PaywallCloseReason): string {
    return reason;
  }

  export function fromJson(json: string): PaywallCloseReason {
    switch (json) {
      case "systemLogic":
        return PaywallCloseReason.SystemLogic;
      case "forNextPaywall":
        return PaywallCloseReason.ForNextPaywall;
      case "webViewFailedToLoad":
        return PaywallCloseReason.WebViewFailedToLoad;
      case "manualClose":
        return PaywallCloseReason.ManualClose;
      case "none":
        return PaywallCloseReason.None;
      default:
        throw new Error(`Invalid PaywallCloseReason value: ${json}`);
    }
  }
}
